src/pytest_mock_resources/fixture/database/generic.py

```diff
@@ -11,9 +11,9 @@ def __init__(self, drivername, host, port, database, username, password):
         self.password = password
 
     def as_url(self):
-        return str(self.as_url())
+        return str(self._as_sqlalchemy_url())
```

Comment:
This seems like a change in behavior is this intentional

Reply:
Yes.
So, initially, `as_url` was just calling itself, which was incorrect, and `_as_sqlalchemy_url()` returns a `URL(...)` object that once converted to a string works across SQL alchemy engines and as well as the Redis client.